src/main/java/dbaccessor/message/MessageAccessor.java

```diff
@@ -42,9 +51,21 @@ public void updateReceivedTs(Message message) {
     /**
      * Completes all DB insertions for the CreateMessage API in a single transaction.
      */
-    public boolean createMessageInTransaction(Message message, Chat chat) {
-        return spannerOperations.performReadWriteTransaction(
+    public boolean createMessageInTransaction(Message message, Chat chat, Optional<Attachment> optionalAttachment) {
+        return spannerOperations.performReadWriteTransaction (
             transactionSpannerOperations -> {
+                if (optionalAttachment.isPresent()) {
+                    Attachment attachment = optionalAttachment.get();
+                    Storage storage = StorageOptions.newBuilder().setProjectId(ApplicationVariable.GCP_PROJECT).build().getService();
+                    BlobId blobId = BlobId.of(ApplicationVariable.GCS_BUCKET, Long.toString(attachment.getAttachmentId()));
+
+                    try {
+                        storage.create(BlobInfo.newBuilder(blobId).build(), attachment.getFile().getBytes());
```

Comment:
Shouldn't uploading the attachment be part of the `AttachmentAccessor` class instead of the `MessageAccessor`.

Reply:
Yes, that would be accurate but I think we would want to keep all accessed for createMessage in a single transaction. I cannot use another `Autowired` variable inside the transaction as it is treated as a `nested` transaction which is not allowed.